src/java.base/share/classes/java/nio/Direct-X-Buffer.java.template

```diff
@@ -78,8 +79,16 @@ class Direct$Type$Buffer$RW$$BO$
         }
 
         public void run() {
-            UNSAFE.freeMemory(address);
-            Bits.unreserveMemory(size, capacity);
+            try {
+                UNSAFE.freeMemory(address);
+                Bits.unreserveMemory(size, capacity);
+            } catch (Throwable x) {
+                // Long-standing behavior: when deallocation fails, VM exits.
+                if (System.err != null) {
+                    new Error("Cleaner terminated abnormally", x).printStackTrace();
```

Comment:
The message would be more useful to identify this as a **Buffer** Cleaner terminated abnormally.

Reply:
This code was cribbed from https://github.com/openjdk/jdk/pull/22165, but your request has led me to
discover that wasn't right. I'm going to need to do some rework of the exception handling around cleaning.